plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/package-rates.tsx

```diff
@@ -68,13 +68,6 @@ const PackageRates = ( {
 		}
 	}, [ onSelectRate, rates, selectedOption ] );
 
-	// Update the data store when the local selected rate changes.
-	useEffect( () => {
-		if ( selectedOption ) {
-			onSelectRate( selectedOption );
-		}
-	}, [ onSelectRate, selectedOption ] );
-
```

Comment:
I feel this change should have been bigger, why didn't we remove the `onSelectRate` and go up even further?

Reply:
What do you mean bigger? There was a duplicate update (this caused 2 requests instead of 1, so for this PR, immediate cancel and retry).